src/luarocks/core/cfg.lua

```diff
@@ -518,7 +518,7 @@ local function make_defaults(lua_version, target_cpu, platforms, home)
       defaults.arch = "netbsd-"..target_cpu
    elseif platforms.solaris then
       defaults.arch = "solaris-"..target_cpu
-      defaults.variables.MAKE = "gmake"
+      defaults.variables.MAKE = os.getenv("MAKE") or "gmake"
```

Comment:
I'm concerned about the changes to the BSD-ish platforms. We are defaulting to `gmake` because Makefile-based rockspecs are mostly written by Linux users who don't stick to "standard POSIX make" limitations. Defaulting to `gmake` in those platforms has proven to be a less bumpy experience for end-users.

Reply:
Do most BSD environments where this will run have a `MAKE` var just floating around that would commonly override this, or would it only be set when set deliberately?

Reply:
If this is ever run from *inside* a Makefile then yes `MAKE` will be set to the make that ran it, which is probably not good.

Reply:
removed the getenv for systems which use gmake